test/unit/mdc-slider/foundation.test.js

```diff
@@ -0,0 +1,728 @@
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({width: 0, left: 0});
+  foundation.init();
+
+  td.verify(mockAdapter.registerInteractionHandler('mousedown', isA(Function)));
+  td.verify(mockAdapter.registerInteractionHandler('pointerdown', isA(Function)));
+  td.verify(mockAdapter.registerInteractionHandler('touchstart', isA(Function)));
+  td.verify(mockAdapter.registerInteractionHandler('keydown', isA(Function)));
+  td.verify(mockAdapter.registerInteractionHandler('focus', isA(Function)));
+  td.verify(mockAdapter.registerInteractionHandler('blur', isA(Function)));
+  td.verify(mockAdapter.registerThumbContainerInteractionHandler('mousedown', isA(Function)));
+  td.verify(mockAdapter.registerThumbContainerInteractionHandler('pointerdown', isA(Function)));
+  td.verify(mockAdapter.registerThumbContainerInteractionHandler('touchstart', isA(Function)));
+  td.verify(mockAdapter.registerResizeHandler(isA(Function)));
+
+  raf.restore();
+});
+
+test('#init performs an initial layout of the component', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+  const {anything} = td.matchers;
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({width: 100, left: 200});
+  foundation.init();
+
+  raf.flush();
+
+  td.verify(mockAdapter.setThumbContainerStyleProperty(anything(), anything()));
+  td.verify(mockAdapter.setTrackStyleProperty(anything(), anything()));
+
+  raf.restore();
+});
+
+test('#destroy deregisters all component event handlers registered during init()', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+
+  foundation.destroy();
+
+  td.verify(mockAdapter.deregisterInteractionHandler('mousedown', isA(Function)));
+  td.verify(mockAdapter.deregisterInteractionHandler('pointerdown', isA(Function)));
+  td.verify(mockAdapter.deregisterInteractionHandler('touchstart', isA(Function)));
+  td.verify(mockAdapter.deregisterInteractionHandler('keydown', isA(Function)));
+  td.verify(mockAdapter.deregisterInteractionHandler('focus', isA(Function)));
+  td.verify(mockAdapter.deregisterInteractionHandler('blur', isA(Function)));
+  td.verify(mockAdapter.deregisterThumbContainerInteractionHandler('mousedown', isA(Function)));
+  td.verify(mockAdapter.deregisterThumbContainerInteractionHandler('pointerdown', isA(Function)));
+  td.verify(mockAdapter.deregisterThumbContainerInteractionHandler('touchstart', isA(Function)));
+  td.verify(mockAdapter.deregisterResizeHandler(isA(Function)));
+});
+
+test('#layout re-computes the bounding rect for the component on each call', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+  let numComputations = 0;
+
+  td.when(mockAdapter.computeBoundingRect()).thenDo(() => {
+    numComputations++;
+    return /* dummy value */ {};
+  });
+
+  foundation.layout();
+  foundation.layout();
+
+  assert.equal(numComputations, 2);
```

Comment:
If you try and `td.verify` a method that's been mocked out, testdouble will emit a warning ([rightfully so](https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md#why-shouldnt-i-call-both-tdwhen-and-tdverify-for-a-single-interaction-with-a-test-double)). Added an explanation as to why we need to stub and verify , thus use this method, in a comment.